libdnf5-cli/output/advisoryinfo.cpp

```diff
@@ -81,17 +96,90 @@ void AdvisoryInfo::Impl::add_advisory(IAdvisory & advisory) {
     }
 }
 
+void AdvisoryInfoJSON::Impl::add_advisory(IAdvisory & advisory) {
+    json_object * json_advisory = json_object_new_object();
+    json_object_object_add(json_advisory, "Name", json_object_new_string(advisory.get_name().c_str()));
+    json_object_object_add(json_advisory, "Title", json_object_new_string(advisory.get_title().c_str()));
+    json_object_object_add(json_advisory, "Severity", json_object_new_string(advisory.get_severity().c_str()));
+    json_object_object_add(json_advisory, "Type", json_object_new_string(advisory.get_type().c_str()));
+    json_object_object_add(json_advisory, "Status", json_object_new_string(advisory.get_status().c_str()));
+    json_object_object_add(json_advisory, "Vendor", json_object_new_string(advisory.get_vendor().c_str()));
+    json_object_object_add(
+        json_advisory,
+        "Issued",
+        json_object_new_string(libdnf5::utils::string::format_epoch(advisory.get_buildtime()).c_str()));
+    json_object_object_add(json_advisory, "Description", json_object_new_string(advisory.get_description().c_str()));
+    json_object_object_add(json_advisory, "Message", json_object_new_string(advisory.get_message().c_str()));
+    json_object_object_add(json_advisory, "Rights", json_object_new_string(advisory.get_rights().c_str()));
+
+    // References
+    json_object * json_references = json_object_new_array();
+    for (auto & reference : advisory.get_references()) {
+        json_object * json_reference = json_object_new_object();
+        json_object_object_add(json_reference, "Title", json_object_new_string(reference->get_title().c_str()));
+        json_object_object_add(json_reference, "Id", json_object_new_string(reference->get_id().c_str()));
+        json_object_object_add(json_reference, "Type", json_object_new_string(reference->get_type_cstring()));
+        json_object_object_add(json_reference, "Url", json_object_new_string(reference->get_url().c_str()));
+        json_object_array_add(json_references, json_reference);
+    }
+    json_object_object_add(json_advisory, "references", json_references);
+
+    // Collections
+    json_object * json_collections = json_object_new_object();
+    for (auto & collection : advisory.get_collections()) {
+        // Modules
+        auto modules = collection->get_modules();
+        if (!modules.empty()) {
+            json_object * json_modules = json_object_new_array();
+            auto module_iter = modules.begin();
+            json_object_array_add(json_modules, json_object_new_string((*module_iter)->get_nsvca().c_str()));
+            module_iter++;
```

Comment:
This is a little bit confusing, I think we don't have to do the first iteration outside of the loop here.
Unlike the regular version the first json line is formatted the same way as the rest.

Reply:
Fixed